server/src/main/kotlin/suwayomi/tachidesk/manga/impl/sync/KoreaderSyncService.kt

```diff
@@ -0,0 +1,359 @@
+                    .select(ChapterTable.manga)
+                    .where { ChapterTable.id eq chapterId }
+                    .firstOrNull()
+                    ?.get(ChapterTable.manga)
+                    ?.value ?: return@transaction null
+
+            val cbzFile = File(getChapterCbzPath(mangaId, chapterId))
+            if (!cbzFile.exists()) {
+                logger.info { "[KOSYNC HASH] Could not generate hash for chapterId=$chapterId. CBZ not found." }
+                return@transaction null
+            }
+
+            val newHash = KoreaderHelper.hashContents(cbzFile)
+            if (newHash != null) {
+                ChapterTable.update({ ChapterTable.id eq chapterId }) {
+                    it[koreaderHash] = newHash
+                }
+                logger.info { "[KOSYNC HASH] Generated and saved new hash for chapterId=$chapterId" }
+            } else {
+                logger.warn { "[KOSYNC HASH] Hashing failed for chapterId=$chapterId." }
+            }
+            newHash
+        }
+    }
+
+    private suspend fun register(
+        username: String,
+        userkey: String,
+    ): AuthResult {
+        val payload =
+            buildJsonObject {
+                put("username", username)
+                put("password", userkey)
+            }
+        val request =
+            buildRequest("${serverConfig.koreaderSyncServerUrl.value.removeSuffix("/")}/users/create") {
+                post(payload.toString().toRequestBody("application/json".toMediaType()))
+            }
+
+        return try {
+            network.client.newCall(request).execute().use { response ->
+                if (response.isSuccessful) {
+                    AuthResult(true, "Registration successful.")
+                } else {
+                    val errorBody = response.body.string()
+                    val errorMessage =
+                        runCatching {
+                            jsonMapper.fromJsonString<Map<String, String>>(
+                                errorBody,
+                                Map::class.java,
+                            )["message"]
+                        }.getOrNull()
+                    val finalMessage = errorMessage ?: "Registration failed with code ${response.code}"
+                    AuthResult(false, finalMessage)
+                }
+            }
+        } catch (e: Exception) {
+            logger.error(e) { "[KOSYNC REGISTER] Exception" }
+            AuthResult(false, e.message)
+        }
+    }
+
+    private suspend fun authorize(
+        username: String,
+        userkey: String,
+    ): AuthResult {
+        val request =
+            buildRequest("${serverConfig.koreaderSyncServerUrl.value.removeSuffix("/")}/users/auth") {
+                get()
+                addHeader("x-auth-user", username)
+                addHeader("x-auth-key", userkey)
+            }
+
+        return try {
+            network.client.newCall(request).execute().use { response ->
```

Comment:
This needs to use .await() instead of .execute(), make sure to change each instance